pdns/bpf-filter.cc

```diff
@@ -320,6 +322,25 @@ BPFFilter::BPFFilter(std::unordered_map<std::string, MapConfiguration>& configs,
     throw std::runtime_error("Unsupported eBPF map format, the current internal implemenation only supports the legacy format");
   }
 
+  struct rlimit old_limit;
+  const size_t new_limit_size = 1073741824;
```

Comment:
Hmmm, if the target is 1024k, should that number be `1048576`? And for the 64k check, `65536` instead of `67108865`? If so, I guess we also need to change the `$USER hard memlock 1073741824` line in the documentation.

Reply:
The rlim_cur and rlim_max returned by getrlimit is a multiplied value of 64k, actually multiplying it by 1024. While what we set in limit.conf is exact value as you mentioned. So I should change in the documentation, but will check in the code what the getrlimit function call returns and why is the value different from what I get from `ulimit -a`

Reply:
`ulimit` uses memory sizes in kbytes. `getrlimit` and `setrlimit` use bytes.

Reply:
I'm confused, are we trying to raise the limit to 1024 kilobytes or more? If we are trying to raise it to 1024 kilobytes, I believe we need to replace `1024 * 1024 * 1024` by `1024 * 1024` and, `64 * 1024 * 1024` by `64 * 1024`, right?